### PR TITLE
Refactored cli arguments

### PR DESCRIPTION
**Pull request by sverhoeven**
Refs #35

Also sneaked in exit=1 when no urls are provided (refs #34)

Can be tested by running

1. `fairtally https://github.com/fair-software/repo1`, will write `tally.html`
2. `fairtally --output-file results.html https://github.com/fair-software/repo1`, will write `results.html`
3. `fairtally --format html https://github.com/fair-software/repo1`, will write `tally.html`
4. `fairtally --format json https://github.com/fair-software/repo1`, will write `tally.json`
5. `fairtally --format json --output-file results.json https://github.com/fair-software/repo1`, will write `results.json`
6. Ultimate unixy command `curl https://research-software.nl/api/software | jq -r '[.[].repositoryURLs.github] | flatten | .[]' | head -3 | fairtally -i - -o - > tally.html`

Only problem I had that when a long URL is checked the completed message leaves trailing remains of URL.

### CODE REVIEW

File: fairtally/cli.py [hunk 1]
```diff
@@ -1,68 +1,47 @@
...
-
-    if output_file_html is not None:
-        write_as_html(results, output_file_html)
-
-
-if __name__ == "__main__":
-    cli()
+    results = [check_url(url, current_value) for url in url_progressbar]
+
+    if output_filename == 'tally.html' and output_format == 'json':
```
Comment (fdiblen) [inline fairtally/cli.py:34]: Can we use a CONSTANT instead of a hard-coded filename?

File: tests/test_cli.py [hunk 2]
```diff
@@ -31,38 +31,81 @@ def mocked_internet():
...
+def invoke_cli(mocked_internet, cli_runner):
     def _invoker(args, **kwargs):
         with mocked_internet:
-            return runner.invoke(cli, args, catch_exceptions=False, **kwargs)
+            return cli_runner.invoke(cli, args, catch_exceptions=False, **kwargs)
 
     return _invoker
 
 
 def test_nourl(invoke_cli):
```
Comment (fdiblen) [inline tests/test_cli.py:47]:
```suggestion
def test_no_url(invoke_cli):
```

File: tests/test_cli.py [hunk 3]
```diff
@@ -31,38 +31,81 @@ def mocked_internet():
...
+    assert "No URLs provided, aborting." in result.stderr
+    assert result.exit_code == 1
+
+
+def read_file(filename):
+    with open(filename, 'r') as f:
+        return f.read()
+
+
+def test_singleurl_nooptions(invoke_cli, cli_runner: CliRunner):
```
Comment (fdiblen) [inline tests/test_cli.py:59]:
```suggestion
def test_singleurl_no_options(invoke_cli, cli_runner: CliRunner):
```

File: tests/test_cli.py [hunk 4]
```diff
@@ -31,38 +31,81 @@ def mocked_internet():
...
-    result = invoke_cli("https://github.com/fair-software/repo1")
-    assert "fairtally progress" in result.output
+        assert "fairtally progress" in result.stderr
+        assert "Completed howfairis checks on 1 URLs results written to tally.html" in result.stderr
+        assert result.exit_code == 0
+        assert url in read_file("tally.html")
 
 
-def test_input_from_file(invoke_cli, tmp_path):
+def test_input_from_file2stdout(invoke_cli, tmp_path):
```
Comment (fdiblen) [inline tests/test_cli.py:70]:
```suggestion
def test_input_from_file_to_stdout(invoke_cli, tmp_path):
```

File: tests/test_cli.py [hunk 5]
```diff
@@ -31,38 +31,81 @@ def mocked_internet():
...
 
-    assert "fairtally progress" in result.output
+    assert "fairtally progress" in result.stderr
+    assert "Completed howfairis checks on 1 URLs results written to -" in result.stderr
+    assert result.exit_code == 0
+    assert "https://github.com/fair-software/repo1" in result.stdout
 
 
-def test_input_from_stdin(invoke_cli):
+def test_input_from_stdin2stdout(invoke_cli):
```
Comment (fdiblen) [inline tests/test_cli.py:82]:
```suggestion
def test_input_from_stdin_to_stdout(invoke_cli):
```

File: tests/test_cli.py [hunk 6]
```diff
@@ -31,38 +31,81 @@ def mocked_internet():
...
+    assert "https://github.com/fair-software/repo1" in result.stdout
+
+
+def test_json_format(invoke_cli, cli_runner):
+    with cli_runner.isolated_filesystem():
+        url = "https://github.com/fair-software/repo1"
+        result = invoke_cli([url, "--format", "json"])
+
+        assert "fairtally progress" in result.stderr
+        assert "Completed howfairis checks on 1 URLs results written to tally.json" in result.stderr
```
Comment (fdiblen) [inline tests/test_cli.py:99]: In the messages can we use `fairtally` instead of `howfairis`?

File: tests/test_cli.py [hunk 7]
```diff
@@ -31,38 +31,81 @@ def mocked_internet():
...
+        assert result.exit_code == 0
+        assert url in read_file("tally.json")
+
+
+def test_json_format2stdout(invoke_cli):
```
Comment (fdiblen) [inline tests/test_cli.py:104]:
```suggestion
def test_json_format_to_stdout(invoke_cli):
```

File: tests/test_cli.py (hunk 6 above)
Comment (sverhoeven) [reply tests/test_cli.py]: Done

File: fairtally/cli.py (hunk 1 above)
Comment (sverhoeven) [reply fairtally/cli.py]: done

**Review by fdiblen — APPROVED** [review]
Amazing @sverhoeven. Good job! :partying_face:

**Pull request merged**